Wabbajack.VirtualFileSystem/Context.cs

```diff
@@ -146,33 +160,11 @@ class Box<T>
             });
         }
 
-        public async Task WriteToFile(string filename)
+        public void WriteToFile()
         {
-            using (var fs = File.Open(filename, FileMode.Create))
-            using (var bw = new BinaryWriter(fs, Encoding.UTF8, true))
-            {
-                fs.SetLength(0);
-
-                bw.Write(Encoding.ASCII.GetBytes(Magic));
-                bw.Write(FileVersion);
-                bw.Write((ulong) Index.AllFiles.Count);
-
-                (await Index.AllFiles
-                    .PMap(Queue, f =>
-                    {
-                        var ms = new MemoryStream();
-                        f.Write(ms);
-                        return ms;
-                    }))
-                    .Do(ms =>
-                    {
-                        var size = ms.Position;
-                        ms.Position = 0;
-                        bw.Write((ulong) size);
-                        ms.CopyTo(fs);
-                    });
-                Utils.Log($"Wrote {fs.Position.ToFileSizeString()} file as vfs cache file {filename}");
-            }
+            CompositeDisposable.Dispose();
```

Comment:
This might not be doing what you think it is.  This disposal will unsubscribe the subscription on line 65, so that it no longer exists or does any work.  It will stop the subscription work immediately (maybe after finishing its current item?), -without- clearing the remaining items on the "queue".

Disposal like this is for when we're all done, and want all the subscriptions cleaned up and destroyed... which I imagine is not what you are after here?

If you're after a "hey, don't' expect any more items.. and finish all that you have" logic.. you'll probably want something like:
```
// Tell the subject/observable that there are no more items coming down the line.
// If/when the last item is processed, the observable is now "complete" and any onComplete callbacks
// will at that point fire.
_nextIndexedFile.Complete();

// We've marked the subject as complete, but there's still a lot of items in the pipe, potentially.
// This can be used to "wait" for all the items to be processed, even if they're all being done on different threads or whatever.
// This of course requires the WriteToFile() to be made an async function again
await _nextIndexedFile;

// Now that we've marked the observable complete, and waited for all the items to be finished... now we can shut down and return
CompositeDisposable.Dispose(); // Destroy all subscription wiring
_vfsBinaryWriter.Close();
_vfsFileStream.Close();